openmc/model/funcs.py

```diff
@@ -442,3 +446,131 @@ def subdivide(surfaces):
         regions.append(+s0 & -s1)
     regions.append(+surfaces[-1])
     return regions
+
+
+def pin(surfaces, materials, subdivisions=None, divide_vols=True,
+        universe_id=None, name=""):
+    """Convenience function for building a fuel pin
+
+    Parameters
+    ----------
+    surfaces : iterable of :class:`openmc.Cylinder`
+        Cylinders used to define boundaries
+        between materials. All cylinders must be
+        concentric and of the same orientation, e.g.
+        all :class:`openmc.ZCylinder`
+    materials : iterable of :class:`openmc.Material`
+        Materials to go between ``surfaces``. There must be one
+        more material than surfaces, corresponding to the material
+        that spans all space outside the final ring.
+    subdivisions : None or dict of int to int
+        Dictionary describing which rings to subdivide and how
+        many times. Keys are indexes of the annular rings
+        to be divided. Will construct equal area rings
+    divide_vols : bool
+        If this evaluates to ``True``, then volumes of subdivided
+        materials will also be divided by the number of divisions.
+        Otherwise the volume of the original material will not be
+        modified before subdivision
+    universe_id : None or int
+        Identifier for this universe
+    name : str
+        Name for this universe
+
+    Returns
+    -------
+    :class:`openmc.Universe`
+        Universe of concentric cylinders filled with the desired
+        materials
+    """
+    check_type("materials",  materials, Iterable, Material)
```

Comment:
Really, this argument should accept anything which is a valid fill. That could be a material. a universe, a lattice, or `None` (indicating no material). It probably suffices to just check that this argument is an iterable, because the `cell.fill` setter will catch any types that don't make sense as a fill.